metals/src/main/scala/scala/meta/internal/metals/CompilerConfiguration.scala

```diff
@@ -162,7 +162,27 @@ class CompilerConfiguration(
     protected def newCompiler(classpath: Seq[Path]): PresentationCompiler = {
       val name = scalaTarget.scalac.getTarget().getUri
       val options = enrichWithReleaseOption(scalaTarget)
-      fromMtags(mtags, options, classpath ++ additionalClasspath, name, search)
+      // val filteredOptions = plugins.filterSupportedOptions(options)
+      // Best Effort option `-Ybest-effort` is useless for PC,
+      // and it may unnecesarily dump semanticdb and tasty files
+      val pcOptions =
+        options.filter(_ != "-Ybest-effort") :+ "-Ywith-best-effort-tasty"
+      val bestEffortDirs = scalaTarget.info
+        .getDependencies()
```

Comment:
After rebasing, I'm having some issues with this scalaTarget.info.getDependencies() method. Supposing I have projects `lib` (no additional dependencies) and `app` (depends on `lib`), when I import the project and I open the `app` module first in vscode, metals creates a newCompiler here for it, but this getDependencies() method returns an empty list (if I first create the newCompiler for `lib` and then `dep`, it works). I wonder if it was the same earlier but I did not notice or if it was changed with the lazier classpath resolution...

Reply:
lazy classpath resolution should not change anything for non-bazel BSP server. At least that would be unexpected. Dependencies should be resolved for all targets at the same time, I don't really see why it would be empty.

Reply:
Ah I found it, the problem turned out to be elsewhere (think I misread my logs). Sorry about that